README.md

```diff
@@ -1,282 +1,257 @@
 ### Import the module(s)
 ```
 import Vault from 'vault-wrapper';
 import { DeviceEventEmitter } from 'react-native';
 ```
-### Authentication
-#### Check Account
-```
-VaultWallet.hasAccount((cb) => {
-  if (cb) {
-    console.log("we have an account")
-  } else {
-    console.log("we dont have an account")
-  }
-})
-```
+### Vault Engine Primitive Concepts
+#### User Accounts 
+A user account is created using a unique identifer generated by some kind of authentication mechanism, here oAUTH. You can think of a user account as the parent to all other data. On the Vault Engine Server a user account is used to associate public keys with human readable names, allowing seamless crpyto user experiences. Additionally it is reccomended that a recover neumonic is generated to be assoicated with the user account. By doing this future private key pairs (Key Accounts) will be recoverable easily by the end user. When a user account is deleted from a device, the sdk deletes all local information assoicated with that user account, such as private/public keys, cached transaction information, etc. 
+
+#### Key Accounts 
+Within the Vault Engine SDK key accounts are the name for a public + private key pair and the associated metadata included along with them. Conceptually, you can think of this as a child object of a User Account, because each key account is assoicated with a User. 
+> * alletAccount` is an object representing the `publicAddress`, `type` and `nickname`**
+
+#### Shared Accounts
+A shared account is type of key account in which the public key is publically avaliable to other app users via a 'connection'. Unlike a default key account, this type of account, and its assoicated stored coins, will be both visable to your connections; as well as the endpoint which others will transact to. For example if two users were to wish to transact using Shared Accounts, each person would see and trasact to each other persons shared account (think abstracted public key). 
+> * haredAccount` is an object representing the `public address`, `type`, `isMyAccount`, `nickname`and `userId`** -- All the accounts you share to others and the accounts shared to you
+#### Connections
+Connections can be thought of as 'friends'. Using this feature of the SDK allows abstraction of public keys and surfacing of simple human readable account names associated defintivly with users. Akin to social features, connections can be sent, recieved, accepted, and denied. A master list of all connection names (users) is stored in the Vault Engine cloud backend so each instance can query a live list.  
+
+### User Accounts and Authentication
 #### Add/Update Account
+Need more details onwhat the account is. Like is this the account that you AUTH into locally, or the 'account' used on the cloud to identify the user? What is jwt, where is userId initalized/come from? On the backend what actions does this call peform for the 'account'? does it generate the wallet for the account etc? 
```

Comment:
when you add an account --> 
The local device will store this token and use it to login in case of any network or power failure
The token is what the server uses to *authenticate* the user/connection and *authorize* access to the Vault Engine. This is simply just allowing the user to log into the vault engine. Remeber user account vs wallet account